InteractionDesignFoundation/nova-html-card
# Add option to include basic styling for HTML elements

## Context

There is no default basic styling applied to HTML elements within `HtmlCard` currently. An `h1`, an `h2`, a `p` and a bulleted list will all look exactly the same. If you write your docs in HTML you could just apply your own classes or add Tailwind classes that are available in Nova, but if you write in Markdown, it's a bit more of a problem. 

## Changes in this PR
I've added some very basic styling for common HTML elements. Users of this package have probably mitigated the lack of styling by adding some basic CSS themselves, so I didn't want to create a breaking change by just adding styles directly. Instead, I've made it opt-in. You can add the styles by doing the following:

```php
(new HtmlCard())->withBasicStyles()->markdown('# Hello World')
```
## Testing

To test this with the IxDF codebase:

1. Add the package as a path repository in `composer.json`:

```json
"repositories": [
       ...
        {
            "type": "path",
            "url": "../nova-html-card" // assuming the package is in the same directory as `ixdf-web` project
        }
    ],
```

2. Run `composer update`
3. In the `construct()` method of `IndexHelpCard`, add `$this->withBasicStyles()`. All of the index page doc cards should now have some styling. 

## Expected feedback

- @mateusjunges Are you happy with this method of adding basic styles? Is there a better way to do it without creating breaking changes?
- @CY8s The styles are loaded in the `CardContent` vue component rather than a separate CSS file due to how Nova loads assets. It's much cheaper to add a bit more JS to an existing file instead of loading a whole new stylesheet. Could you please check if there are any missing common styles or improvements that could be made? I used the GH markdown styling as a base for font sizes and margins.
